File: backend/src/main/java/com/example/backend/model/dto/ApplicationUserDTO.java
package com.example.backend.model.dto;

import com.example.backend.model.Role;

import java.util.HashSet;
import java.util.Set;

public class ApplicationUserDTO {
    private Long id;
    private String name;
    private String email;
    private String password;
    private Role role;
    private Set<Long> classRoomIds = new HashSet<>();

    private Set<Long> commitmentIds = new HashSet<>();

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public Role getRole() {
        return role;
    }

    public void setRole(Role role) {
        this.role = role;
    }

    public Set<Long> getClassRoomIds() {
        return classRoomIds;
    }

    public void setClassRoomIds(Set<Long> classRoomIds) {
        this.classRoomIds = classRoomIds;
    }

    public Set<Long> getCommitmentIds() {
        return commitmentIds;
    }

    public void setCommitmentIds(Set<Long> commitmentIds) {
        this.commitmentIds = commitmentIds;
    }
}

File: frontend/src/app/shared/dto/application-user.dto.ts
import {RoleEnum} from "../enum/role.enum";

export class ApplicationUserDto {
  public readonly id: number | null;
  public readonly name: string;
  public readonly email: string;
  public readonly password: string;
  public readonly role: RoleEnum;
  public readonly profilePictureId: number | null;
  public readonly classRoomIds: Set<number>;
  public readonly commitmentIds: Set<number>;


  constructor(
    id: number | null,
    name: string,
    email: string,
    password: string,
    role: RoleEnum,
    profilePictureId: number | null,
    classRoomIds: Set<number>,
    commitmentIds: Set<number>
  ) {
    this.id = id;
    this.name = name;
    this.email = email;
    this.password = password;
    this.role = role;
    this.profilePictureId = profilePictureId;
    this.classRoomIds = classRoomIds;
    this.commitmentIds = commitmentIds;
  }
}
